etl/steps/data/garden/covid/latest/cases_deaths.py

```diff
@@ -0,0 +1,512 @@
+    # the daily change to NA. This also sets the 7-day rolling average to NA for the next 7 days.
+    tb.loc[tb["new_cases"] < 0, "new_cases"] = np.nan
+    tb.loc[tb["new_deaths"] < 0, "new_deaths"] = np.nan
+
+    # Custom data corrections
+    for ldc in LARGE_DATA_CORRECTIONS:
+        tb.loc[(tb["country"] == ldc[0]) & (tb["date"].astype(str) == ldc[1]), f"new_{ldc[2]}"] = np.nan
+
+    for ldc in LARGE_DATA_CORRECTIONS_SINCE:
+        tb.loc[(tb["country"] == ldc[0]) & (tb["date"].astype(str) >= ldc[1]), f"new_{ldc[2]}"] = np.nan
+
+    # Sort (legacy)
+    tb = tb.sort_values(["country", "date"])
+
+    return tb
+
+
+def add_regions(tb: Table, ds_regions: Dataset, ds_income: Dataset) -> Table:
+    tb = geo.add_regions_to_table(
+        tb,
+        ds_regions,
+        ds_income,
+        year_col="date",
+        regions={
+            # Standard continents
+            "Africa": {},
+            "Asia": {},
+            "Europe": {},
+            "North America": {},
+            "Oceania": {},
+            "South America": {},
+            # Income groups
+            "Low-income countries": {},
+            "Lower-middle-income countries": {},
+            "Upper-middle-income countries": {},
+            "High-income countries": {},
+            # Special regions
+            "European Union (27)": {},
+            "World excl. China": {
+                "additional_regions": ["Asia", "Africa", "Europe", "North America", "Oceania", "South America"],
+                "excluded_members": ["China"],
+            },
+            "World excl. China and South Korea": {
+                "additional_regions": ["Asia", "Africa", "Europe", "North America", "Oceania", "South America"],
+                "excluded_members": ["China", "South Korea"],
+            },
+            "World excl. China, South Korea, Japan and Singapore": {
+                "additional_regions": ["Asia", "Africa", "Europe", "North America", "Oceania", "South America"],
+                "excluded_members": ["China", "South Korea", "Japan", "Singapore"],
+            },
+            "Asia excl. China": {
+                "additional_regions": ["Asia"],
+                "excluded_members": ["China"],
+            },
+        },
+    )
+    tb = geo.add_regions_to_table(
+        tb,
+        ds_regions,
+        ds_income,
+        year_col="date",
+        regions={"World": {}},
+    )
+    return tb
+
+
+def add_period_aggregates(tb: Table, prefix: str, periods: int):
+    # Convert Table to DataFrame
+    df = pd.DataFrame(tb)
+
+    # Period-aggregate cases and deaths
+    cases_colname = f"{prefix}_cases"
+    deaths_colname = f"{prefix}_deaths"
+    df[[cases_colname, deaths_colname]] = (
+        df[["country", "new_cases", "new_deaths"]]
+        .groupby("country")[["new_cases", "new_deaths"]]
+        .rolling(window=periods, min_periods=periods - 1, center=False)
```

Comment:
I'll see if I can make these work with Table (but feel free to merge it like this).